test/jdk/java/nio/channels/FileChannel/Transfer.java

```diff
@@ -158,6 +159,30 @@ public void testReadableByteChannel() throws Exception {
         }
     }
 
+    @Test
+    public void transferToTrusted() throws IOException { // for bug 8325382
+        File source = File.createTempFile("before", "after");
+        source.deleteOnExit();
+
+        CharSequence csq = "Reality is greater than the sum of its parts.";
+        Files.writeString(source.toPath(), csq);
+        final long length = csq.length();
+        assert source.length() == length;
+
+        File target = File.createTempFile("before", "after");
+        target.deleteOnExit();
+
+        try (FileInputStream in = new FileInputStream(source);
+             FileOutputStream out = new FileOutputStream(target);
+             FileChannel chSource = in.getChannel();
+             FileChannel chTarget = out.getChannel()) {
+            // The count of bytes requested to transfer must exceed
+            // FileChannelImpl.MAPPED_TRANSFER_THRESHOLD which is
+            // currently 16384
+            chSource.transferTo(length, 16385, chTarget);
```

Comment:
"transferToTrusted" hints that this method will test a trusted target but it's actually a very specific test that doesn't test all trusted scenarios, and doesn't even check the return from transferTo. So I think this one is really a regression test that checks that it doesn't throw.

Reply:
Attempted to fix in ecd01954da61240c20143bba5348c7cba4d1a2f4.